### PR TITLE
move to cargo clippy instead of using the clippy plugin

### LINKED ISSUE #729: Cannot publish serde_derive
```
$ git checkout v0.9.3
  HEAD is now at 8624ca6... Release 0.9.3

$ cd serde_derive

$ cargo publish
      Updating registry `https://github.com/rust-lang/crates.io-index`
     Packaging serde_derive v0.9.3 (file://../serde_derive)
     Verifying serde_derive v0.9.3 (file://../serde_derive)
      Updating registry `https://github.com/rust-lang/crates.io-index`
  error: failed to verify package tarball

  Caused by:
    cyclic package dependency: package `cargo_metadata v0.1.1` depends on itself

$ cargo tree --features clippy
  serde_derive v0.9.3 (file://../serde_derive)
  ├── clippy v0.0.112
  │   ├── cargo_metadata v0.1.1
  │   │   ├── serde v0.9.3
  │   │   ├── serde_derive v0.9.2
  │   │   │   ├── quote v0.3.12
  │   │   │   ├── serde_codegen_internals v0.12.0
  │   │   │   │   └── syn v0.11.4
  │   │   │   │       ├── quote v0.3.12 (*)
  │   │   │   │       └── unicode-xid v0.0.4
  │   │   │   └── syn v0.11.4 (*)
  │   │   └── serde_json v0.9.1
  │   │       ├── dtoa v0.3.1
  │   │       ├── itoa v0.2.1
  │   │       ├── num-traits v0.1.36
  │   │       └── serde v0.9.3 (*)
  │   └── clippy_lints v0.0.112
  │       ├── matches v0.1.4
  │       ├── quine-mc_cluskey v0.2.4
  │       ├── regex-syntax v0.4.0
  │       ├── semver v0.2.3
  │       │   └── nom v1.2.4
  │       ├── toml v0.2.1
  │       │   └── rustc-serialize v0.3.22
  │       └── unicode-normalization v0.1.3
  ├── quote v0.3.12 (*)
  ├── serde_codegen_internals v0.12.0 (file://../serde_codegen_internals)
  │   └── syn v0.11.4 (*)
  └── syn v0.11.4 (*)
```

@oli-obk any ideas? Serde_derive depends on clippy which depends on cargo_metadata which depends on serde_derive.

### PR DESCRIPTION
**Pull request by oli-obk**
fixes #729 

let's see what travis says

### CODE REVIEW

**Comment by oli-obk** [comment]
screenshot from 2017-01-31 12-54-28

**Comment by oli-obk** [comment]
I set it up so clippy failing to build doesn't show up as a build failure

File: travis.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+else
+    (cd serde && travis-cargo build)
+    (cd serde && travis-cargo --only beta test)
+    (cd serde && travis-cargo --only nightly test -- --features unstable-testing)
+    (cd serde && travis-cargo build -- --no-default-features)
+    (cd serde && travis-cargo --only nightly build -- --no-default-features --features alloc)
+    (cd serde && travis-cargo --only nightly build -- --no-default-features --features collections)
+    (cd test_suite && travis-cargo --only beta test)
+    (cd test_suite/deps && travis-cargo --only nightly build)
+    (cd test_suite travis-cargo --only nightly test -- --features unstable-testing)
```
Comment (dtolnay) [inline travis.sh]: This sneakily does not run the test suite :scream:. I'll fix it.

Comment (dtolnay) [inline travis.sh]: Fixed in #735.

Comment (oli-obk) [inline travis.sh]: Underhanded bash is too easy

**Pull request merged**